Robots/goalkeeper_controller.py

```diff
@@ -0,0 +1,41 @@
+import numpy as np
+import math
+from GameEngine.ball_model import BallModel
+from GameEngine.robot_model import RobotModel
+from Robots.robot_control import Robot, Goal
+
+
+class GoalkeeperController(Robot):
+
+    def __init__(self):
```

Comment:
reference to the robot model could be passed in object initialisation, probably same with ball (not in `accrue_sensors_data`)

..Thus, .I'm not sure if this is right approach; so you can keep current structure and we can get back to it another time